ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/fragment/AEMArticleItemFragment.java

```diff
@@ -138,21 +147,62 @@ private String injectCss(@NonNull final String content) {
 
     // endregion WebView content
 
+    //region WebClient
+
     private class AEMWebViewClient extends WebViewClient {
         @Nullable
         @Override
         public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
-            if (url.endsWith(".jpg")) {
-                //TODO: Load local Image WebResource
-            } else if (url.endsWith(".css")) {
-                //TODO: Load CSS WebResource
+            String extension = MimeTypeMap.getFileExtensionFromUrl(url);
+
+            if (extension.isEmpty()) {
+                return super.shouldInterceptRequest(view, url);
+            }
+            try {
+
+                String mimeType;
+
+                switch (extension) {
+                    case "jpg":
+                    case "png":
+                    case "bmp":
+                    case "gif":
+                        mimeType = String.format("image/%s", "jpg".equals(extension) ? "jpeg" : extension);
+                        return getResponseFromFile(mimeType, url);
+                    case "css":
+                        mimeType = "text/css";
+                        return getResponseFromFile(mimeType, url);
+                }
+            } catch (FileNotFoundException e) {
+                Timber.d(e);
             }
 
             return super.shouldInterceptRequest(view, url);
         }
+
+        private WebResourceResponse getResponseFromFile(@NonNull String mimeType, @NonNull String url)
+                throws FileNotFoundException {
+
+            Resource resource = ArticleRoomDatabase.getInstance(requireActivity()).resourceDao()
```

Comment:
we use the `ArticleRoomDatabase` in multiple locations, let's centralize getting the singleton in `onCreate()`.
so:
```java
public void onCreate() {
   super.onCreate();
   mAemDb = ArticleRoomDatabase.getInstance(requireContext());
```

and then this becomes `mAemDb.resoruceDao().find(Uri.parse(url))`

Reply:
good catch